### ISSUE TITLE
Pipeline crashes, but no errors are reported in the HTML report

### ISSUE THREAD

**Issue by mgxd** [poldracklab/fmriprep#595]
just like poldracklab/mriqc#434

**Comment by chrisfilo** [comment]
Are the errors printed at the end of the HTML reports not good enough?

**Comment by effigies** [comment]
I think it's a pretty reasonable default, given we encourage people to use Docker.

**Comment by chrisfilo** [comment]
I agree - I just wanted to make sure @mgxd knows about the error printouts in the HTML reports. Maybe we don't advertise it well enough.

**Comment by mgxd** [comment]
@chrisfilo the error is happening during the `epi_mask_report`, so I don't see any generated html file

**Comment by effigies** [comment]
Hmm. The report should be generated, even if it the pipeline doesn't finish and it's missing pieces.

**Comment by mgxd** [comment]
@effigies sorry, was looking in the out directory. I can find reports in the working directory!

**Comment by effigies** [comment]
So completed reports should appear in the working directory, but the compiled report `sub-<subject_label>.html` should appear in the output directory, at the same level as the `sub-<subject_label/`.

**Comment by mgxd** [comment]
In that case, the compiled report is missing. In my `/outdir/sub-{subject}`, I have just `anat`, `func`, and `log` directories.

**Comment by chrisfilo** [comment]
The compiled reports should be in `<out_dir>/fmriprep/sub-{subject}.html`

**Comment by chrisfilo** [comment]
Just to clarify - did you find the compiled reports in `<out_dir>/fmriprep/sub-{subject}.html` or are they missing?

If they are missing could you share the console output of that execution?

**Comment by mgxd** [comment]
@chrisfilo they are missing.

Here's the [fat log file](https://www.dropbox.com/s/mrx1p38lg8o053v/fmriprep.log?dl=0)

**Comment by chrisfilo** [comment]
```
170720-17:03:53,512 workflow INFO:
	 [Job finished] jobname: _epi_to_mni_transform138 jobid: 19368
170720-17:03:54,953 workflow INFO:
	 Executing: _epi_to_mni_transform148 ID: 19378
170720-17:03:55,135 workflow INFO:
	 Executing node _epi_to_mni_transform148 in dir: /mnt/workdir/mathias/fmriprepwork/fmriprep_wf/single_subject_BANDA001_wf/func_preproc_task_rest_run_03_wf/epi_mni_trans_wf/epi_to_mni_transform/mapflow/_epi_to_mni_transform148
170720-17:03:55,157 workflow INFO:
	 Running: antsApplyTransforms --default-value 0 --float 1 --input /mnt/workdir/mathias/fmriprepwork/fmriprep_wf/single_subject_BANDA001_wf/func_preproc_task_rest_run_03_wf/epi_hmc_wf/split/vol0148.nii.gz --interpolation LanczosWindowedSinc --output vol0148_trans.nii.gz --reference-image /mnt/workdir/mathias/fmriprepwork/fmriprep_wf/single_subject_BANDA001_wf/func_preproc_task_rest_run_03_wf/epi_mni_trans_wf/gen_ref/1mm_T1_reference.nii.gz --transform /mnt/workdir/mathias/fmriprepwork/fmriprep_wf/single_subject_BANDA001_wf/anat_preproc_wf/t1_2_mni/ants_t1_to_mniComposite.h5 --transform /mnt/workdir/mathias/fmriprepwork/fmriprep_wf/single_subject_BANDA001_wf/func_preproc_task_rest_run_03_wf/epi_reg_wf/fsl2itk_fwd/affine.txt --transform /mnt/workdir/mathias/fmriprepwork/fmriprep_wf/single_subject_BANDA001_wf/func_preproc_task_rest_run_03_wf/epi_hmc_wf/hcm2itk/mapflow/_hcm2itk148/affine.txt
slurmstepd: error: *** JOB 9128989 ON node013 CANCELLED AT 2017-07-20T13:03:58 DUE TO TIME LIMIT ***
```
Is this the correct log? What time limit did you set when submitting the job?

**Comment by effigies** [comment]
```
170719-17:10:55,740 workflow INFO:
	 Executing: ds_epi_mask_report ID: 0
170719-17:10:55,743 workflow INFO:
	 Executing node ds_epi_mask_report in dir: /mnt/workdir/mathias/fmriprepwork/fmriprep_wf/single_subject_BANDA001_wf/func_preproc_task_rest_run_04_wf/func_reports_wf/ds_epi_mask_report
170719-17:10:55,778 workflow ERROR:
	 ['Node ds_epi_mask_report failed to run on host node013.cm.cluster.']
```

What pipeline plugin are you using? It looks like these derivative nodes are being run out-of-order. Or somehow you're hitting a case where the connections aren't getting set properly.

Also, the job being cancelled explains the lack of report. The reports are built after the workflow exits.

**Comment by mgxd** [comment]
@chrisfilo yes - I set it to run for 1 day, but there are many functionals/high res scans so I'll re-run with more time.

@effigies ah I see. I've running the default, which I thought was MultiProc? Essentially, my call is:
```
sbatch -t 1-00:00:00 --mem 20GB singularity exec -B $base:/mnt -c docker://poldracklab/fmriprep:latest fmriprep $DATADIR $OUTDIR participant --participant_label $subj --nthreads 10 --mem_mb 20000 --no-freesurfer --ignore slicetiming -w $WORKDIR
```
perhaps there's a better way to submit?

**Comment by effigies** [comment]
It should be MultiProc.

**Comment by mgxd** [comment]
@effigies here's the generated crashfile

```
Node Name: fmriprep_wf.single_subject_BANDA001_wf.func_preproc_task_rest_run_04_wf.func_reports_wf.ds_epi_mask_report
File: crash-20170725-152740-mathiasg-ds_epi_mask_report-dcbec612-71f9-42d1-9318-f9f468b3a09a.pklz
Working Directory: /mnt/workdir/mathias/fmriprepwork/fmriprep_wf/single_subject_BANDA001_wf/func_preproc_task_rest_run_04_wf/func_reports_wf/ds_epi_mask_report
Inputs: 
base_directory: /mnt/workdir/mathias/fmriprepwork/reportlets
extra_values:
ignore_exception: False
in_file:
source_file: /mnt/data/sub-BANDA001/func/sub-BANDA001_task-rest_run-04_bold.nii.gz
suffix: epi_mask

Traceback (most recent call last):

  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/plugins/multiproc.py", line 322, in _send_procs_to_workers
    self.procs[jobid].run()
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 372, in run
    self._run_interface()
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 482, in _run_interface
    self._result = self._run_command(execute)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 613, in _run_command
    result = self._interface.run()
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base.py", line 1066, in run
    self._check_mandatory_inputs()
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base.py", line 971, in _check_mandatory_inputs
    raise ValueError(msg)
ValueError: DerivativesDataSink requires a value for input 'in_file'. For a list of required inputs, see DerivativesDataSink.help()
```

**Comment by chrisfilo** [comment]
This situation (pipeline crashes, but no errors are reported in the HTML report) is still happening. check out https://openneuro.org/datasets/ds001040/versions/00001

**Comment by oesteban** [comment]
@chrisfilo is this still an issue on that dataset?

**Comment by chrisfilo** [comment]
That dataset was deleted in the meantime - let's assume it's solved via #690

**Comment by effigies** [comment]
Please reopen if not solved.
